File: Software Development Module/Week4/ShopApp.java
/*
ShopApp.java
Billy Cussen
10/07/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;

public class ShopApp {
    
    public static void main(String[]args){

        //Variables
        int hoodies = 0;
        int tshirts = 0;
        int caps = 0;
        double total;
        boolean inputCheckShop;

        //Objects
        Shop myShop;
        Scanner keyboard;

        //Input
        do{
            try{
                inputCheckShop = true;
                keyboard = new Scanner(System.in);
                System.out.print("How many hoodies would you like: ");
                hoodies = keyboard.nextInt();
            }catch(InputMismatchException e){
                System.out.println("Please enter the number of Hoodies you would like(e.g. 1)");
                inputCheckShop = false;
            }
        }while(!inputCheckShop);

        do{
            try{
                inputCheckShop = true;
                keyboard = new Scanner(System.in);
                System.out.print("How many tshirts would you like: ");
                tshirts = keyboard.nextInt();
            }catch(InputMismatchException e){
                System.out.println("Please enter the number of Tshirts you would like(e.g. 1)");
                inputCheckShop = false;
            }
        }while(!inputCheckShop);

        do{
            try{
                inputCheckShop = true;
                keyboard = new Scanner(System.in);
                System.out.print("How many caps would you like: ");
                caps = keyboard.nextInt();
            }catch(InputMismatchException e){
                System.out.println("Please enter the number of Caps you would like(e.g. 1)");
                inputCheckShop = false;
            }
        }while(!inputCheckShop);

        //Compute
        keyboard = null;
        myShop = new Shop(hoodies,tshirts,caps);
        myShop.computeTotal();

        //Output
        total = myShop.getTotal();
        System.out.println("The total of your purchase is: "+total);
    }

}

File: Python Course/Software Development Module - in Python/Week4/ShopApp.py
"""
ShopApp.py
Billy Cussen
27/09/2020
"""

#Objects
import gc
import Shop
from Shop import myShop

#Variables
repeat = "y"

while repeat == "y":

    inputCheck = True

    while inputCheck:
        try:
            hoodies = int(input("Please enter the Number of Hoodies being Purchased: "))
        except:
            print("You need to enter an Integer (e.g. 1)")
        else:
            if hoodies < 0:
                print("You need to enter a number greater than 0")
            else:
                inputCheck = False

    while inputCheck == False:
        try:
            tshirts = int(input("Please enter the Number of Tshirts being Purchased: "))
        except:
            print("You need to enter an Integer (e.g. 1)")
        else:
            if tshirts < 0:
                print("You need to enter a number greater than 0")
            else:
                inputCheck = True

    while inputCheck:
        try:
            caps = int(input("Please enter the Number of Caps being Purchased: "))
        except:
            print("You need to enter an Integer (e.g. 1)")
        else:
            if caps < 0:
                print("You need to enter a number greater than 0")
            else:
                inputCheck = False

    myShopApp = myShop(hoodies,tshirts,caps)
    total = myShopApp.computeTotal()
    myShopApp = None
    gc.collect()
    print ("The total of this purchase is: "+str(total))

    repeat = ""
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter if you would like to repeat(\"Y\" or \"N\"): "))
        repeat = repeat.lower()
        if repeat != "y" and repeat != "n":
            print("Please enter \"Y\" or \"N\"")
    
print ("Thanks for using my App!")